engine/crates/composition/src/compose/context.rs

```diff
@@ -42,8 +42,13 @@ impl<'a> Context<'a> {
         self.ir.insert_enum(name)
     }
 
-    pub(crate) fn insert_enum_value(&mut self, enum_id: federated::EnumId, value: StringWalker<'_>) {
-        self.ir.insert_enum_value(enum_id, value);
+    pub(crate) fn insert_enum_value(
+        &mut self,
+        enum_id: federated::EnumId,
+        value: StringWalker<'_>,
+        deprecation: Option<Option<StringWalker<'_>>>,
```

Comment:
That should probably be an `Option<Deprecated>` with `struct Deprecated<'a> { reason: Option<StringWalker<'a>> }`, ping if you think that would be clearer and I'll change that.

Reply:
Yeah would be clearer :)